pkg/domain/infra/abi/containers.go

```diff
@@ -147,11 +147,40 @@ func (ic *ContainerEngine) ContainerUnpause(ctx context.Context, namesOrIds []st
 	return report, nil
 }
 func (ic *ContainerEngine) ContainerStop(ctx context.Context, namesOrIds []string, options entities.StopOptions) ([]*entities.StopReport, error) {
-	names := namesOrIds
-	ctrs, rawInputs, err := getContainersAndInputByContext(options.All, options.Latest, names, ic.Libpod)
+	containersNamesOrIds := namesOrIds
+	if len(options.Filters) > 0 {
+		filterFuncs := make([]libpod.ContainerFilter, 0, len(options.Filters))
+		if len(options.Filters) > 0 {
+			for k, v := range options.Filters {
+				generatedFunc, err := dfilters.GenerateContainerFilterFuncs(k, v, ic.Libpod)
+				if err != nil {
+					return nil, err
+				}
+				filterFuncs = append(filterFuncs, generatedFunc)
+			}
+		}
+		candidates, err := ic.Libpod.GetContainers(filterFuncs...)
+		if err != nil {
+			return nil, err
+		}
+		containersNamesOrIds = []string{}
+		for _, candidate := range candidates {
+			if options.All {
+				containersNamesOrIds = append(containersNamesOrIds, candidate.ID())
+				continue
+			}
+			for _, nameOrID := range namesOrIds {
+				if nameOrID == candidate.ID() || nameOrID == candidate.Name() {
+					containersNamesOrIds = append(containersNamesOrIds, nameOrID)
+				}
+			}
+		}
+	}
```

Comment:
I am convinced that this logic exists somewhere else already.  Maybe we can reuse parts of `pkg/ps` or integrate it into `getContainersAndInputByContext`?